golang/cmd/sensorconnect/processSensorData.go

```diff
@@ -0,0 +1,450 @@
+	if !reflect.DeepEqual(item.SimpleDatatype.Type, ""){ // enters if simple datatype
+		if  (reflect.DeepEqual(item.SimpleDatatype.ValueRange, "") && reflect.DeepEqual(item.SimpleDatatype.SingleValue, "")){ // simple datatype doesn't contain SingleValue or ValueRange
+			return nil, nil
+		}else if !reflect.DeepEqual(item.SimpleDatatype.ValueRange, ""){// simple datatype contains ValueRange
+			if reflect.DeepEqual(datatype, "IntegerT"){
+				intUpperBound, err := strconv.Atoi(item.SimpleDatatype.ValueRange.UpperValue)
+				if err != nil {
+					return
+				}
+				intLowerBound, err := strconv.Atoi(item.SimpleDatatype.ValueRange.LowerValue)
+				if err != nil {
+					return
+				}
+			} else if reflect.DeepEqual(datatype, "UIntegerT"){
+				intUpperBound, err := strconv.Atoi(item.SimpleDatatype.ValueRange.UpperValue)
+				if err != nil {
+					return
+				}
+				intLowerBound, err := strconv.Atoi(item.SimpleDatatype.ValueRange.LowerValue)
+				if err != nil {
+					return
+				}
+			} else if reflect.DeepEqual(datatype, "Float32T"){
+				floatUpperBound, err := strconv.ParseFloat(item.SimpleDatatype.ValueRange.UpperValue, 32)
+				if err != nil {
+					return
+				}
+				floatLowerBound, err := strconv.ParseFloat(item.SimpleDatatype.ValueRange.UpperValue, 32)
+				if err != nil {
+					return
+				}
+			return nil, checkIfValueInValueRange()
+		}
+	}
+
+	}
+	} else if !reflect.DeepEqual(item.DatatypeRef.DatatypeId, "") { // true if record item includes a datatypeRef -> look for type into DatatypeCollection with id
+		for _, datatypeElement := range datatypeArray {
+			if reflect.DeepEqual(datatypeElement.Id, item.DatatypeRef.DatatypeId) {
+
+				return
+			}
+		}
+		err = errors.New("DatatypeRef.DatatypeId is not in DatatypeCollection of Iodd file -> Datatype could not be determined.")
+		return
+	} else {
+		err = errors.New("Neither SimpleDatatype nor DatatypeRef included in Recorditem")
+		return
+}
+
+
+// valueRangeOrSingleValueExists returns true if a RecordItem holds single values or valueRanges
+func valueRangeOrSingleValueExistsInSimpleDatatype(item RecordItem){
+	if (reflect.DeepEqual(item.SimpleDatatype.ValueRange, "") && reflect.DeepEqual(item.SimpleDatatype.SingleValue, ""){
+		return false
+	} else{
+		return true
+	}
+}
+*/
+// convertBinaryValueToString converts a binary string value to a readable string according to its datatype
+func convertBinaryValueToString(binaryValue string, datatype string) (output string) {
+	switch datatype {
+	case "OctetStringT":
+		output = BinToHex(binaryValue)
+	default:
+		outputString, _ := strconv.ParseUint(binaryValue, 2, 64)
+		return fmt.Sprintf("%v", outputString)
+	}
+	return
+}
+
+// createDigitalInputPayload creates a json output body from a DigitalInput to send via mqtt or kafka to the server
+func createDigitalInputPayload(serialNumber string, portNumberString string, timestampMs string, dataPin2In []byte) (payload []byte) {
+	payload = []byte(`{
```

Comment:
serial_number is not required. you can remove it entirely

Reply:
fixed